src/commands/heroku.rs

```diff
@@ -390,6 +392,152 @@ pub fn restart_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandR
     Ok(())
 }
 
+#[command]
+#[num_args(3)]
+pub fn deploy_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandResult {
+    let app_name = args
+        .single::<String>()
+        .expect("You must include an app name");
+
+    let source_url = args
+        .single::<String>()
+        .expect("You must include a link to a tar file containing the code you want to deploy");
+
+    let app_version = args
+        .single::<String>()
+        .expect("You must include an app version");
+
+    let buildpack_params = heroku_rs::endpoints::builds::post::BuildpackParam {
+        url: String::from("https://github.com/emk/heroku-buildpack-rust"),
+        name: String::from("emk/rust"),
+    };
```

Comment:
This turned out to be much more interesting than I anticipated.

To create a build of an application, you need to pass the buildpack information (as described above). 

For the Rust buildpack, the name is "emk/rust" and the url is "https://github.com/emk/heroku-buildpack-rust. In order to create a build using that buildpack, I need to pass those values to the api call to create the build. 

```bash
$ curl -n -X POST https://api.heroku.com/apps/$APP_ID_OR_NAME/builds \
  -d '{
  "buildpacks": [
    {
      "url": "https://github.com/heroku/heroku-buildpack-ruby",
      "name": "heroku/ruby"
    }
  ],
```

However, I found something odd when I requested the existing buildpack information for an app and tried to use that to create a new build of that app.

When I requested information about the buildpacks already installed on the app, I received this response:

```bash
nell@nsharrington-45404 $ curl -n https://api.heroku.com/apps/polar-ravine-00412/buildpack-installations -H "Accept: application/vnd.heroku+json; version=3"                                       [                                                                                                                  {                                                                                                                  "buildpack":{                                                                                                      "url":"https://buildpack-registry.s3.amazonaws.com/buildpacks/emk/rust.tgz",                                     "name":"https://buildpack-registry.s3.amazonaws.com/buildpacks/emk/rust.tgz"                                   },                                                                                                               "ordinal":0                                                                                                    }                                                                                                              ]  
```

Note that it returns the S3 address of the buildpack for both the url and the name of the buildpack (I confirmed this with raw api calls as well as calls formatted using the heroku-rs crate).

If I try to pass a struct containing those values when creating a new build, it errors out.

```bash
 !    error fetching custom buildpack https://buildpack-registry.s3.amazonaws.com/buildpacks/https://buildpack-registry.s3.amazonaws.com/buildpacks/emk/rust.tgz.tgz

 !     Push failed
```

So....I don't think pulling the existing app buildpack information through the Heroku API, then passing it to a create request is going to work. However, I can definitely make the buildpack(s) configurable through an environmental variable. I will try that approach.

Reply:
Actually - I just discovered that if you don't pass a buildpack to the create request it will just use whatever buildpacks are already installed for the app. Problem solved!